Tasks/DownloadBuildArtifactsV0/main.ts

```diff
@@ -265,68 +265,80 @@ async function main(): Promise<void> {
-                        console.log(tl.loc("DownloadArtifacts", artifact.name, itemsUrl));
-
-                        var variables = {};
-                        var webProvider = new providers.WebProvider(itemsUrl, templatePath, variables, handler);
-                        var fileSystemProvider = new providers.FilesystemProvider(downloadPath);
-
-                        downloadPromises.push(downloader.processItems(webProvider, fileSystemProvider, downloaderOptions).catch((reason) => {
+                        const operationName: string = `downloadContainer- ${artifact.name}`;
+                        const config: IContainerHandlerConfig = {
+                            artifactInfo: artifact,
+                            downloadPath: downloadPath,
+                            downloaderOptions: downloaderOptions,
+                            checkDownloadedFiles: checkDownloadedFiles,
+                            endpointUrl: endpointUrl,
+                            templatePath: templatePath,
+                            handler: handler
+                        };
+
+                        const downloadHandler: DownloadHandlerContainer = new DownloadHandlerContainer(config);
+                        const downloadPromise: Promise<models.ArtifactDownloadTicket[]> = executeWithRetries(
+                            operationName,
+                            () => downloadHandler.downloadResources(),
+                            retryLimit
+                        ).catch((reason) => {
                             reject(reason);
-                        }));
+                            return;
+                        });
+
+                        downloadPromises.push(downloadPromise);
+                        await downloadPromise;
                     }
                 }
                 else if (artifact.resource.type.toLowerCase() === "filepath") {
-                    let downloader = new engine.ArtifactEngine();
-                    let downloadUrl = artifact.resource.data;
-                    let artifactName = artifact.name.replace('/', '\\');
-                    let artifactLocation = path.join(downloadUrl, artifactName);
-                    if (!fs.existsSync(artifactLocation)) {
-                        console.log(tl.loc("ArtifactNameDirectoryNotFound", artifactLocation, downloadUrl));
-                        artifactLocation = downloadUrl;
-                    }
-
-                    console.log(tl.loc("DownloadArtifacts", artifact.name, artifactLocation));
-                    var fileShareProvider = new providers.FilesystemProvider(artifactLocation, artifactName);
-                    var fileSystemProvider = new providers.FilesystemProvider(downloadPath);
-
-                    downloadPromises.push(downloader.processItems(fileShareProvider, fileSystemProvider, downloaderOptions).catch((reason) => {
+                    const operationName: string = `downloadByFilePath- ${artifact.name}`;
```

Comment:
This also seems to be the same for all cases

Reply:
The operation name is different for each download case. I changed these names to make this more obvious. 
Thanks!